praw/models/reddit/reasons.py

```diff
@@ -0,0 +1,170 @@
+    ======================= ===================================================
+    """
+
+    STR_FIELD = "id"
+
+    def __eq__(self, other):
+        """Return whether the other instance equals the current."""
+        if isinstance(other, str):
+            return other == str(self)
+        return (
+            isinstance(other, self.__class__)
+            and str(self) == str(other)
+            and other.subreddit == self.subreddit
+        )
+
+    def __hash__(self):
+        """Return the hash of the current instance."""
+        return (
+            hash(self.__class__.__name__)
+            ^ hash(str(self))
+            ^ hash(self.subreddit)
+        )
+
+    def __init__(self, reddit, subreddit, reason_id, _data=None):
+        """Construct an instance of the Reason object."""
+        self.id = reason_id
+        self.subreddit = subreddit
+        super(Reason, self).__init__(reddit, _data=_data)
+
+    def _fetch(self):
+        for reason in self.subreddit.reasons:
+            if reason.id == self.id:
+                self.__dict__.update(reason.__dict__)
+                self._fetched = True
+                return
+        raise ClientException(
+            "/r/{} does not have the reason {}".format(self.subreddit, self.id)
+        )
+
+
+class SubredditReasons:
+    """Provide a set of functions to a Subreddit's removal reasons."""
+
+    def __getitem__(self, reason_id):
+        """Lazily return the Reason for the subreddit with id ``reason_id``.
+
+        :param reason_id: The id of the removal reason
+
+        This method is to be used to fetch a specific removal reason, like so:
+
+        .. code:: python
+
+           reason = reddit.subreddit('praw_test').reasons['141vv5c16py7d']
+           print(reason)
+
+        """
+        return Reason(self._reddit, self.subreddit, reason_id)
+
+    def __init__(self, subreddit):
+        """Create a SubredditReasons instance.
+
+        :param subreddit: The subreddit whose removal reasons to work with.
+
+        """
+        self.subreddit = subreddit
+        self._reddit = subreddit._reddit
+
+    def __iter__(self):
+        """Return a list of Removal Reasons for the subreddit.
+
+        This method is used to discover all removal reasons for a
+        subreddit:
+
+        .. code-block:: python
+
+           for reason in reddit.subreddit('NAME').reasons:
+               print(reason)
+
+        """
+        response = self.subreddit._reddit.get(
+            API_PATH["removal_reasons_list"].format(subreddit=self.subreddit)
+        )
+        for reason_id, reason_data in response["data"].items():
+            yield Reason(
+                self._reddit, self.subreddit, reason_id, _data=reason_data
+            )
+
+    def add(self, title, message):
+        """Add a removal reason to this subreddit.
+
+        :param title: The title of the removal reason
+        :param message: The message to send the user.
```

Comment:
Would it be more appropriate to say, `the message associated with the removal` since by itself this message isn't actually sent to the user?